### PR TITLE
Upgrade go version and other fixes

### PR DESCRIPTION
**Pull request by noornee**
- upgrade golang version
- fix golangci-lint errors
- setup redis docker

### CODE REVIEW

File: src/store/redis.go [hunk 1]
```diff
@@ -53,14 +52,11 @@ func (r *Redis) Connect() error {
 		Password: r.ConnectionInfo.Password,
 		Username: r.ConnectionInfo.Username,
 		DB:       0, // use default DB,
-		TLSConfig: &tls.Config{
-			MinVersion: tls.VersionTLS12,
-		},
```
Comment (odetolakehinde) [inline src/store/redis.go:58]: why? we need it to connect to the stuff in render o

File: redis/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+FROM redis
+COPY redis.conf /usr/local/etc/redis/redis.conf
+CMD [ "redis-server", "/usr/local/etc/redis/redis.conf" ]
```
Comment (odetolakehinde) [inline redis/Dockerfile:3]: add extra line after

**Review by odetolakehinde — CHANGES_REQUESTED** [review]

File: src/store/redis.go (hunk 1 above)
Comment (noornee) [reply src/store/redis.go:58]: wow okay, i was having issues connecting the server to the redis port on docker that's why i removed that line.
i was getting `context deadline exceeded` error

i dont really think the specification of minimum version is required. i've connected to redis on render before without specifying that

Comment (odetolakehinde) [reply src/store/redis.go:58]: have you connected to the one I sent you without this? i need to ensure it's working

Comment (noornee) [reply src/store/redis.go:58]: true that. lemme try connecting with it

Comment (noornee) [reply src/store/redis.go:58]: so it didnt work :smiling_face_with_tear:  
pushing a fix. is okay if i add an env like `REDIS_TLS_ENABLED`?

Comment (odetolakehinde) [reply src/store/redis.go]: yes yes. let's do that

Comment (noornee) [reply src/store/redis.go]: https://github.com/odetolakehinde/slack-stickers-be/pull/13/commits/cd5b2117ce7b225e5ae9a92f87f4e7618c848edc

**Review by odetolakehinde — APPROVED** [review]

**Pull request merged**